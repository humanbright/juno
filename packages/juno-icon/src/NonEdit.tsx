import React, { forwardRef, memo } from 'react';

const NonEdit = memo(
  forwardRef(
    (
      props: React.SVGProps<SVGSVGElement>,
      svgRef?: React.Ref<SVGSVGElement>,
    ) => (
      <svg
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
        ref={svgRef}
        {...props}
      >
        <path d="M2.565 3.979a1 1 0 011.32-.083l.094.083 24.042 24.042a.999.999 0 01-1.32 1.497l-.094-.083L2.565 5.393a.999.999 0 010-1.414zm9.197 10.611l1.414 1.414-6.579 6.58 2.828 2.828 6.579-6.579 1.414 1.414-6.659 6.659c-.279.279-.635.47-1.022.547l-5.006 1.001a1 1 0 01-1.177-1.176l1.001-5.006c.077-.387.268-.743.547-1.022l6.658-6.659zM28.01 4a4 4 0 010 5.657l-7.763 7.762-1.414-1.414 4.935-4.934-2.828-2.828-4.935 4.934-1.414-1.414 7.762-7.762a4 4 0 015.657 0zm-4.243 1.414l-1.413 1.413 2.828 2.828 1.413-1.414a2 2 0 10-2.828-2.828z" />
      </svg>
    ),
  ),
);
NonEdit.displayName = 'NonEdit';
NonEdit['iconName'] = 'non-edit';
export default NonEdit;
