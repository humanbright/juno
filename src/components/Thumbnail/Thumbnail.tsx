import React, { forwardRef, HTMLAttributes, memo } from 'react';

import {
  RcBaseSize,
  styled,
  withDeprecatedCheck,
  useThemeProps,
} from '../../foundation';
import { RcIcon, RcIconProps } from '../Icon';
import { StyledThumbnail } from './styles';

type RcThumbnailSize = RcBaseSize<'small' | 'large'>;

type RcThumbnailProps = {
  /** Thumbnail background url */
  src?: string;
  /** thumbnail size, default is `large` */
  size?: RcThumbnailSize;
  /** that will render RcIcon when not have url */
  symbol?: RcIconProps['symbol'];
  /** @deprecated please change to symbol */
  iconType?: string;
  /** @deprecated Thumbnail background url, please use src to replace that */
  url?: string;
} & HTMLAttributes<HTMLSpanElement>;

const _RcThumbnail = forwardRef<any, RcThumbnailProps>(
  (inProps: RcThumbnailProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcThumbnail' });
    const { url, src = url, iconType, size, symbol, ...rest } = props;
    return (
      <>
        {src ? (
          <StyledThumbnail
            ref={ref}
            src={src}
            size={size}
            data-test-automation-id="thumbnail"
            {...rest}
          />
        ) : (
          <RcIcon
            ref={ref}
            data-test-automation-id="iconThumbnail"
            {...rest}
            size={size === 'small' ? 'small' : 'xxxlarge'}
            symbol={symbol}
          >
            {iconType}
          </RcIcon>
        )}
      </>
    );
  },
);

const RcThumbnail = styled(
  withDeprecatedCheck(
    memo(_RcThumbnail),
    [
      {
        prop: 'iconType',
        time: '2021-2',
        comment: 'Please use symbol to set icon',
      },
      {
        prop: 'url',
        time: '2021-2',
        comment: 'Please use src to set url',
      },
    ],
    'RcThumbnail',
  ),
)``;

RcThumbnail.defaultProps = {
  size: 'large',
};

RcThumbnail.displayName = 'RcThumbnail';

export { RcThumbnail, RcThumbnailProps, RcThumbnailSize };
