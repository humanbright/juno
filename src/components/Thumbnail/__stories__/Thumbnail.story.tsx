import { Meta, Story } from '@storybook/react';
import React, { ComponentProps } from 'react';

import NonEdit from '../../../icon/NonEdit';
import { notShowInDocTable } from '../../../storybook';
import { RcThumbnail } from '../Thumbnail';

export default {
  title: 'ðŸš€ Cleanup Components/Thumbnail',
  component: RcThumbnail,
} as Meta;

type ThumbnailProps = ComponentProps<typeof RcThumbnail>;

export const Thumbnail: Story<ThumbnailProps> = ({ ...args }) => {
  return <RcThumbnail {...args} />;
};

Thumbnail.args = {
  src: 'https://picsum.photos/100/100/?image=1',
  symbol: NonEdit,
};

Thumbnail.argTypes = {
  ...notShowInDocTable<keyof ThumbnailProps>(['iconType', 'url']),
};
