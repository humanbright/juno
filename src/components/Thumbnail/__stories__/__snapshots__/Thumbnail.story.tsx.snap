// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Thumbnail Thumbnail 1`] = `
.c0 {
  width: 36px;
  height: 36px;
  background-image: url(https://picsum.photos/100/100/?image=1);
  border-radius: 4px;
  display: inline-block;
  background-color: #F3F3F3;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}

<span
  className="c0 "
  data-test-automation-id="thumbnail"
  size="large"
  src="https://picsum.photos/100/100/?image=1"
/>
`;
